lib/Hooks/ImageHooks.php

```diff
@@ -49,23 +51,10 @@ public function register() {
 	public function postCreate(Node $node) {
 		$absolutePath = $this->dataDirectory.$node->getPath();
 
-		$dimensions = getimagesize($absolutePath);
+		$metadataExtracter = new ExtractMetadata($absolutePath);
```

Comment:
@oparoz Can you give a basic idea on how to implement DI over here? I am not able to figure out what the interface will be.